src/ui/src/builder/settings/BuilderFieldsWriterResourceId.vue

```diff
@@ -149,20 +151,25 @@ const selected = computed<string | string[]>({
 	},
 });
 
-watch(selected, async (newAppId) => {
-	if (props.resourceType !== "application" || !newAppId) return;
-
-	const appData = selectorEl.value?.selectedData;
-	const inputsList = appData?.inputs;
-
-	if (!inputsList || typeof inputsList !== "object") return;
+const selectedData = ref<WriterApplication | undefined>();
+
+function onSelectedData(appData: WriterApplication | undefined) {
+	selectedData.value = appData; // for readability
```

Comment:
I don't understand why we need to hold this value in the memory. Can't it be
```suggestion
function onSelectedData(appData: WriterApplication | undefined) {
```